### PR TITLE
domain: fix data race in the ttlJobManager

### LINKED ISSUE #41288: Data race in TTL
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_check2/detail/ghpr_check2/13822/pipeline/226

### 2. What did you expect to see? (Required)
no data race

### 3. What did you see instead (Required)

```
==================
WARNING: DATA RACE
Read at 0x00c00d998028 by goroutine 74353:
  runtime.raceread()
      <autogenerated>:1 +0x24
  go.etcd.io/etcd/client/v3/namespace.(*watcherPrefix).Watch()
      external/io_etcd_go_etcd_client_v3/namespace/watch.go:53 +0x4a5
  github.com/pingcap/tidb/ttl/client.(*etcdClient).WatchCommand.func1()
      ttl/client/command.go:258 +0x1ba

Previous write at 0x00c00d998028 by goroutine 71952:
  runtime.racewrite()
      <autogenerated>:1 +0x24
  go.etcd.io/etcd/client/v3/namespace.(*watcherPrefix).Close()
      external/io_etcd_go_etcd_client_v3/namespace/watch.go:81 +0x89
  go.etcd.io/etcd/client/v3.(*Client).Close()
      external/io_etcd_go_etcd_client_v3/client.go:144 +0x8d
  github.com/pingcap/tidb/domain.(*Domain).Close()
      domain/domain.go:907 +0x186
  github.com/pingcap/tidb/tests/realtikvtest.CreateMockStoreAndDomainAndSetup.func2()
      tests/realtikvtest/testkit.go:142 +0x4f
  testing.(*common).Cleanup.func1()
      GOROOT/src/testing/testing.go:1041 +0x193
  testing.(*common).runCleanup()
      GOROOT/src/testing/testing.go:1210 +0x143
  testing.tRunner.func2()
      GOROOT/src/testing/testing.go:1440 +0x52
  runtime.deferreturn()
      GOROOT/src/runtime/panic.go:476 +0x32
  github.com/pingcap/tidb/testkit.(*TestKit).ExecWithContext()
      testkit/testkit.go:323 +0x96e
  github.com/pingcap/tidb/testkit.(*TestKit).MustExecWithContext()
      testkit/testkit.go:132 +0xb7
  github.com/pingcap/tidb/testkit.(*TestKit).MustExec()
      testkit/testkit.go:127 +0xf7
  github.com/pingcap/tidb/tests/realtikvtest.CreateMockStoreAndDomainAndSetup()
      tests/realtikvtest/testkit.go:121 +0x34e
  github.com/pingcap/tidb/sessionctx/variable.glob..func173()
      sessionctx/variable/sysvar.go:898 +0x5e
  github.com/pingcap/tidb/sessionctx/variable.(*SysVar).ValidateWithRelaxedValidation()
      sessionctx/variable/variable.go:361 +0x1c7
  github.com/pingcap/tidb/domain.(*Domain).rebuildSysVarCache()
      domain/sysvar_cache.go:146 +0x8c4
  github.com/pingcap/tidb/domain.(*Domain).LoadSysVarCacheLoop()
      domain/domain.go:1428 +0xa8
  github.com/pingcap/tidb/session.BootstrapSession()
      session/session.go:3320 +0x6d3
  github.com/pingcap/tidb/domain.(*Domain).GetSessionCache()
      domain/sysvar_cache.go:62 +0x59
  github.com/pingcap/tidb/session.(*session).loadCommonGlobalVariablesIfNeeded()
      session/session.go:3639 +0x104
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      session/session.go:2118 +0x145
  github.com/pingcap/tidb/session.(*session).ExecuteInternal()
      session/session.go:1659 +0x31b
  github.com/pingcap/tidb/domain.(*Domain).LoadPrivilegeLoop()
      domain/domain.go:1372 +0x130
  github.com/pingcap/tidb/session.BootstrapSession()
      session/session.go:3313 +0x684
  github.com/pingcap/tidb/tests/realtikvtest.CreateMockStoreAndDomainAndSetup()
      tests/realtikvtest/testkit.go:115 +0x1e4
  github.com/pingcap/tidb/tests/realtikvtest.CreateMockStoreAndSetup()
      tests/realtikvtest/testkit.go:90 +0x52
  tests/realtikvtest/txntest/txntest_test.TestBasicTxnState()
      tests/realtikvtest/txntest/txn_state_test.go:33 +0x3e
  testing.tRunner()
      GOROOT/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      GOROOT/src/testing/testing.go:1493 +0x47
```

### 4. What is your TiDB version? (Required)

https://github.com/pingcap/tidb/pull/41093/commits/03cb1c67a8a0928d0ceeb270ef598db209f8d43c

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #41288

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -902,6 +903,19 @@ func (do *Domain) Close() {
 		do.info.RemoveServerInfo()
 		do.info.RemoveMinStartTS()
 	}
+	ttlJobManager := do.ttlJobManager.Load()
+	if ttlJobManager != nil {
+		ttlJobManager.Stop()
+		err := ttlJobManager.WaitStopped(context.Background(), func() time.Duration {
+			if intest.InTest {
+				return 10 * time.Second
```
Comment (hawkingrei) [inline domain/domain.go:911]: timeout is too long for the test. so it is changed to 10s.

**Review by lcwangchao — APPROVED** [review]

**Comment by YangKeao** [comment]
Then the log in https://github.com/pingcap/tidb/pull/41289/files#diff-6e8dbf8bdec841e376c480ee699a935f12f0edc2b4fed42907edd8cbe676dd74R2550-R2552 is not correct. The `ttlJobManager` didn't exit there.

**Comment by YangKeao** [comment]
> Then the log in https://github.com/pingcap/tidb/pull/41289/files#diff-6e8dbf8bdec841e376c480ee699a935f12f0edc2b4fed42907edd8cbe676dd74R2550-R2552 is not correct. The `ttlJobManager` didn't exit there.

Oops. I realized that the `do.exit` is closed after the `Close` function was called. It's still correct.

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**